File: AspNet472CSharpMVC5/Startup.cs
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
using Microsoft.Owin;
using Microsoft.Owin.Host.SystemWeb;
using Microsoft.Owin.Security.Cookies;

using Owin;

namespace AspNet472CSharpMVC5
{
    public partial class Startup
    {
        public void Configuration(IAppBuilder app)
        {
            app.UseCookieAuthentication(new CookieAuthenticationOptions
            {
                CookieSameSite = SameSiteMode.None,
                CookieHttpOnly = true,
                CookieSecure = CookieSecureOption.Always,
                CookieManager = new SameSiteCookieManager(new SystemWebCookieManager())
            });
        }
    }
}

File: AspNet472VisualBasicMVC5/Startup.vb
' Copyright (c) .NET Foundation. All rights reserved.
' Licensed under the Apache License, Version 2.0. See License.txt In the project root for license information.

Imports Microsoft.Owin.Host.SystemWeb
Imports Microsoft.Owin.Security.Cookies
Imports Owin

Public Class Startup
    Public Sub Configuration(app As IAppBuilder)
        app.UseCookieAuthentication(New CookieAuthenticationOptions() With {
            .CookieSameSite = SameSiteMode.None,
            .CookieHttpOnly = True,
            .CookieSecure = CookieSecureOption.Always,
            .CookieManager = New SameSiteCookieManager(New SystemWebCookieManager())
        })
    End Sub
End Class
